### ISSUE TITLE
False positive for rule Proj1200

### ISSUE THREAD

**Issue by Silvex78** [dotnet-project-file-analyzers/dotnet-project-file-analyzers#422]
The nuget package SkiaSharp.NativeAssets.Linux is needed for some graphical rendering in Linux. The rule states that I need to mark the assemblies as private. If you mark the reference as private the code does not work anymore. Can you investigate why this warning is reported?

**Comment by CptWesley** [comment]
Odd, I haven't ran it through a debugger yet, but the nuspec file does not appear to explicity mark whether or not it's a development dependency, so `IsDevelopmentDependency` should be `null`. The `.nupkg` file appears have empty `lib` subdirs, but the `runtime` subdirectories do contains `.dll`s, so I would expect `HasRuntimeDll` to be `true` which should result in `Proj1200` not triggering.

However, my guess is that the contents of the `runtimes` subdirs might only be extracted from the `.nupkg` file if the target runtime matches that of the system doing the compilation, which means they might not be detected. In this case I would expect this warning not to appear when doing the build on a Linux system (perhaps in some CI pipeline).

I'm not entirely sure what the desired fix for this should be (we could look inside the actual `.nupkg` file if we can resolve it, but I'm not necessarily looking forward to doing that). @Corniel thoughts?

**Comment by Corniel** [comment]
both the extracted files on disc, as the `.nupkg` seem to contain the following dirs and files.
``` tree
nuget\packages\skiasharp.nativeassets.linux\3.116.1
│   .nupkg.metadata
│   .signature.p7s
│   icon.png
│   LICENSE.txt
│   README.md
│   skiasharp.nativeassets.linux.3.116.1.nupkg
│   skiasharp.nativeassets.linux.3.116.1.nupkg.sha512
│   skiasharp.nativeassets.linux.nuspec
│   THIRD-PARTY-NOTICES.txt
│
├───buildTransitive
│   └───net462
│           SkiaSharp.NativeAssets.Linux.targets
│
├───lib
│   ├───net462
│   │       _._
│   │
│   ├───net6.0
│   │       _._
│   │
│   ├───net8.0
│   │       _._
│   │
│   ├───netstandard2.0
│   │       _._
│   │
│   └───netstandard2.1
│           _._
│
└───runtimes
    ├───linux-arm
    │   └───native
    │           libSkiaSharp.so
    │           libSkiaSharp.so.116.0.0
    │
    ├───linux-arm64
    │   └───native
    │           libSkiaSharp.so
    │           libSkiaSharp.so.116.0.0
    │
    ├───linux-musl-x64
    │   └───native
    │           libSkiaSharp.so
    │           libSkiaSharp.so.116.0.0
    │
    └───linux-x64
        └───native
                libSkiaSharp.so
                libSkiaSharp.so.116.0.0
```

I've also no clue what is needed runtime yet. Might be worth to checkout what [SkiaSharp](https://github.com/mono/SkiaSharp) is doing while packing.
